### PR TITLE
Add Azure DevOps Code Example Issue #2562

### PR DESCRIPTION
**Pull request by sunkickr**
Add Azure DevOps code example to CI/CD documentation

### CODE REVIEW

File: cloud/stable/03_deploy/06_ci-cd.md [hunk 1]
```diff
@@ -365,6 +365,29 @@ jobs:
...
+steps:
+- script: |
+    echo "Building container.."
+    docker build -t registry.gcp0001.us-east4.astronomer.io/extraterrestrial-aperature-9667/airflow:$CI_PIPELINE_IID .
+    docker login registry.gcp0001.us-east4.astronomer.io -u _ -p $(SERVICE-ACCOUNT-KEY)
+    docker push registry.gcp0001.us-east4.astronomer.io/extraterrestrial-aperature-9667/airflow:$CI_PIPELINE_IID
+
+```
+
 > **Note:** Make sure to replace `infrared-photon-7780` in the example above with your deployment's release name and to store your Service Account Key in your GitHub repo's secrets according to [this GitHub guide]( https://help.github.com/en/articles/virtual-environments-for-github-actions#creating-and-using-secrets-encrypted-variables).
```
Comment (paolaperaza) [inline cloud/stable/03_deploy/06_ci-cd.md:391]: @sunkickr This "Note" re: `infrared-photon-7780` belongs under the `## GitHub Actions CI/CD` header. Can you pull it back up there?

File: cloud/stable/03_deploy/06_ci-cd.md [hunk 2]
```diff
@@ -365,6 +365,29 @@ jobs:
...
+  vmImage: 'ubuntu-latest'
+ 
+variables:
+- group: Variable-Group
+- group: Key-Vault-Group
+steps:
+- script: |
+    echo "Building container.."
+    docker build -t registry.gcp0001.us-east4.astronomer.io/extraterrestrial-aperature-9667/airflow:$CI_PIPELINE_IID .
+    docker login registry.gcp0001.us-east4.astronomer.io -u _ -p $(SERVICE-ACCOUNT-KEY)
```
Comment (paolaperaza) [inline cloud/stable/03_deploy/06_ci-cd.md:386]:
```suggestion
    docker login registry.gcp0001.us-east4.astronomer.io -u _ -p $(DEPLOYMENT-SERVICE-ACCOUNT-KEY)
```

File: cloud/stable/03_deploy/06_ci-cd.md [hunk 3]
```diff
@@ -365,6 +365,29 @@ jobs:
...
+pool:
+  vmImage: 'ubuntu-latest'
+ 
+variables:
+- group: Variable-Group
+- group: Key-Vault-Group
+steps:
+- script: |
+    echo "Building container.."
+    docker build -t registry.gcp0001.us-east4.astronomer.io/extraterrestrial-aperature-9667/airflow:$CI_PIPELINE_IID .
```
Comment (paolaperaza) [inline cloud/stable/03_deploy/06_ci-cd.md:385]: Typo on `$CI_PIPELINE_IID .`? Should it be `$ CI_PIPELINE_ID`? Is the `.` there correct?

File: cloud/stable/03_deploy/06_ci-cd.md [hunk 4]
```diff
@@ -365,6 +365,29 @@ jobs:
         registry: registry.gcp0001.us-east4.astronomer.io
 ```
 
+## Azure DevOps
+
+The following example uses Azure Key Vault to store secretes as varible groups avalible to the pipeline.
```
Comment (paolaperaza) [inline cloud/stable/03_deploy/06_ci-cd.md:370]: @sunkickr Maybe rephrase to:

```suggestion
This example uses [Azure Key Vault](https://azure.microsoft.com/en-us/services/key-vault/) to store secrets as variable groups that are then made available to all DAGs within an Airflow Deployment on Astronomer.
```

Comment (paolaperaza) [reply cloud/stable/03_deploy/06_ci-cd.md:370]: Where does this yaml script go?

**Review by paolaperaza — CHANGES_REQUESTED** [review]
@sunkickr Thank you for tackling! Left some questions/comments.

File: cloud/stable/03_deploy/06_ci-cd.md (hunk 4 above)
Comment (sunkickr) [reply cloud/stable/03_deploy/06_ci-cd.md:370]: The azure key vault secrets are made available to azure DevOps which has the yaml code that runs the pipeline

File: cloud/stable/03_deploy/06_ci-cd.md (hunk 1 above)
Comment (sunkickr) [reply cloud/stable/03_deploy/06_ci-cd.md:391]: I can fix this

File: cloud/stable/03_deploy/06_ci-cd.md (hunk 3 above)
Comment (sunkickr) [reply cloud/stable/03_deploy/06_ci-cd.md:385]: The `.` is needed because the image is held in a remote repository

File: cloud/stable/03_deploy/06_ci-cd.md (hunk 2 above)
Comment (sunkickr) [reply cloud/stable/03_deploy/06_ci-cd.md:386]: I'll make this change

File: .spelling [hunk 5]
```diff
@@ -4,6 +4,12 @@
 # one word per line, to define a file override use ' - filename'
 # where filename is relative to this configuration file
 # <!-- markdownlint-disable-file -->
+Pre-Requisites
```
Comment (paolaperaza) [inline .spelling:7]:
```suggestion
Prerequisites
```

I think `Pre-Requisites` is actually wrong hahaha whoops

File: cloud/stable/03_deploy/06_ci-cd.md (hunk 3 above)
Comment (paolaperaza) [reply cloud/stable/03_deploy/06_ci-cd.md:387]: Two `I`s in `IID`?

**Review by paolaperaza — APPROVED** [review]
Looks good @sunkickr ! Two small comments on the spelling page and then just double checking there's no typo on something, but approving preemptively. Merge whenever you're ready!

File: cloud/stable/03_deploy/06_ci-cd.md (hunk 3 above)
Comment (sunkickr) [reply cloud/stable/03_deploy/06_ci-cd.md]: The second I isn't needed I can take it out

File: .spelling (hunk 5 above)
Comment (sunkickr) [reply .spelling]: I'll fix this

**Comment by sunkickr** [comment]
Added CI/CD pipeline code, fixed spelling and markdown errors
